src/script/sign.cpp

```diff
@@ -26,7 +26,7 @@ bool TransactionSignatureCreator::CreateSig(std::vector<unsigned char>& vchSig,
 
     uint256 hash;
     try {
-        hash = SignatureHash(scriptCode, *txTo, nIn, nHashType, amount, sigversion);
```

Comment:
This removal makes the  `sigversion` parameter of `CreateSig` unused.
We could remove it from the whole call stack: `Sign1`, `SignN`, and finally `SignStep` (which is also always using  `SIGVERSION_BASE`).

Reply:
good eye, done in b4c8c8e